perfact/zodbsync/helpers.py

```diff
@@ -28,7 +28,7 @@ def read_pdata(obj):
         data = obj.data
         source = ''
         while data is not None:
-            source += data.data
+            source += bytes_to_str(data.data)
```

Comment:
hm this way `source` might end up with a different type than the (linked) data found in the object. Not sure about the consequences down the line, but I think it would be better to fix line 29 instead, initializing `source` to either `u''` or `b''` depending on the type of `data.data`.
However, I am not sure what to do what we should return if `data` is `None`.

Reply:
Better this way? Inititalize source depending on `data.data` type.

My test case btw is to create an empty Zope4 instance, put jquery.min.js (as file) into `Data.fs` and run perfact-zoperecord against it. Worked well using the following:

```python
def read_pdata(obj):
    '''Avoid authentication problems when reading linked pdata.'''
    if isinstance(obj.data, (bytes, unicode)):
        source = obj.data
    else:
        data = obj.data
        if isinstance(data.data, bytes):
            source = b''
        elif isinstance(data.data, str):
            source = ''
        while data is not None:
            source += data.data
            data = data.next
    return source
```

I also tried to handle the `data is None` case, but returning something like an empty string resulted in strange recursion errors.

I would like to use zodbsync inside an automatically built zope4 environment (using zodbsync master), this error keeps me from uploading the first Zope 4 example apps as zodbsync repository here on github.

Reply:
Ah, forget my comment regarding `data` being `None`. I misread the `while` loop as an `if` statement, thinking it can happen that `data` is `None` initially. However, this is only a loop ending condition I guess.

LGTM